File: python/basics/if_statement.py
#!/usr/bin/env python3

# Demonstrating if statements in Python

number = -5
print("Using simple if statement:")
if number > 0:
    print(f"{number} is positive")
elif number < 0:
    print(f"{number} is negative")
else:
    print(f"{number} is zero")

# Using a ternary operator
print("\nUsing ternary operator:")
result = f"{number} is positive" if number > 0 else f"{number} is negative" if number < 0 else f"{number} is zero"
print(result)

File: perl/basics/if_statement.pl
#!/bin/perl

use strict;
use warnings;
use feature 'say';

# This is a simple if statement example in Perl
# It checks if a number is positive, negative, or zero
say "Using sinple if statement:";
my $number = -5;
if ($number > 0) {
    say "$number is positive";
} elsif ($number < 0) {
    say "$number is negative";
} else {
    say "$number is zero";
}
# The above code will output: -5 is negative
# The if statement checks the condition and executes the corresponding block of code
# The elsif statement checks the next condition if the first one is false
# The else statement executes if all previous conditions are false

# You can also use a ternary operator for a shorter version
say "Using ternary operator:";
my $result = ($number > 0) ? "$number is positive" : ($number < 0) ? "$number is negative" : "$number is zero";
say $result;

# The ternary operator is a shorthand for if-else statements
# It evaluates the condition and returns the corresponding value based on the result

